packages/edit-post/src/components/header/index.js

```diff
@@ -51,7 +51,10 @@ function Header( {
 						forceIsSaving={ isSaving }
 					/>
 				) }
-				<PostPreviewButton />
+				<PostPreviewButton
+					forceIsAutosaveable={ hasActiveMetaboxes }
+					forcePreviewLink={ isSaving ? null : undefined }
```

Comment:
it can't because we're forcing the `null` value from outside. The inner component don't know what value to put unless we rename the prop something like `forceEmptyPreviewLink`